### PR TITLE
[StructuralMechanicsApplication][Release] Clean up cpp warnings

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR removes several warnings from cpp tests in StructuralMechanicsApplication

Migrates #4197 to the Release-7.0

### CODE REVIEW

**Comment by philbucher** [comment]
does this resolve the tolerance issue?

**Comment by roigcarlo** [comment]
@loumalouomega Seems the test took to long :S

**Comment by loumalouomega** [comment]
@roigcarlo I don't understand the error. This should not be an issue. This is already in the master

**Comment by roigcarlo** [comment]
There is no error :S its just the tests are taking too long. 
The execution time is only 1 minute for all small tests, and take into account that by reducing the compile time we have increased a little bit the execution time of some tests, specially the big ones.

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp [hunk 1]
```diff
@@ -134,7 +134,7 @@ SolvingStrategyType::Pointer CreatePrimalSolvingStrategy(ModelPart* pModelPart)
     SchemeType::Pointer p_scheme =
         Kratos::make_shared<ResidualBasedBossakDisplacementScheme<SparseSpaceType, LocalSpaceType>>(0.0);
     ConvergenceCriteriaType::Pointer p_conv_criteria =
-        Kratos::make_shared<ResidualCriteria<SparseSpaceType, LocalSpaceType>>(1e-24, 1e-25);
+        Kratos::make_shared<ResidualCriteria<SparseSpaceType, LocalSpaceType>>(1e-12, 1e-14);
```
Comment (roigcarlo) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp]: @loumalouomega Could this cause a difference in the execution time?

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp]: All the contrary, the number of iterations will decrease. In fact that was the origin of the origin PR. That the cpp tests did not converge due to a tight convergence tolerance

Comment (roigcarlo) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp]: Then I have no clue.

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp]: Sincerely I don't know

**Comment by loumalouomega** [comment]
> There is no error :S its just the tests are taking too long.
> The execution time is only 1 minute for all small tests, and take into account that by reducing the compile time we have increased a little bit the execution time of some tests, specially the big ones.

I am lost with this

**Comment by philbucher** [comment]
I will take a look tonight

**Comment by philbucher** [comment]
I give up
There seems to be a significant mismatch btw master and release now, I lost the overview of the patches and fixes that were applied :/

**Comment by loumalouomega** [comment]
I think I will close this. Too much smoke on the Release

**Pull request closed without merging**